server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/list/ListTasksResponse.java

```diff
@@ -204,14 +235,14 @@ public XContentBuilder toXContentGroupedByNone(XContentBuilder builder, Params p
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
-        toXContentGroupedByParents(builder, params);
+        toXContentGroupedByNone(builder, params);
```

Comment:
I overlooked this in my first round of reviews, I'd like to undestand why this needs changing.

Reply:
as far as I understand this makes parsing easier, yet there is the concern that it may break existing transport client users calling `toXContent` on the obtained response object, given that the method is public. I wonder if we could add parsing without this slight breakage and how much work that would be. It is odd that we need this behaviour change technically only  to make our tests work, given that `fromXContent` knows to parse only the `groupedByNone` output. What options do we have?

- make fromXContent parse the groupedByParents output instead?
- make it possible in some way to call toXContentGroupedByNone in our tests without changing the default behaviour of toXContent. This may be possible by making the output depend on a param passed using the params argument. The test would also have to set that param, which is not supported at the moment. This sounds rather complicated, not sure it' worth it
- Accepting this slight breaking change

Other alternatives?

Reply:
@javanna I hear you, if non-breaking changes is that important then better find solution without it.

Another alternative - don't change default serialization and use custom testcase. Like it is now in this PR in `ListTasksResponseTests::testXContentWithFailures()`.
So, in this testcase we can explicitly call `ListTasksResponse::toXContentGroupedByNone`, parse, do object comparison, etc.
Not the best solution also though.

Reply:
Could we move toXContent to look at params and call the proper method depending on that? Could the REST action call that instead of the custom rest builder listeners? Then we could reuse your test changes from the cluster health PR, where you can define ToXContent.Params through a new protected method and provide the one that we need in the test? Not sure if I am missing something, I haven't tried this out but this could be the way to go.

Reply:
@javanna It is nice idea, but I stuck with implementation - `toXContentGroupedByNode` requires `DiscoveryNodes` parameter, which cannot be passed as `Params` bc `Params` can store only `String` values.
We can add `paramAsObject` and add implementation which can store any `Object` but not sure that it is a good idea. What do you think?

Reply:
I see the problem @Van0SS thanks for bringing that up. I am thinking that it's not wort to refactor this for the corner case that we are worrying about.

Reply:
@javanna Absolutely :)